### PR TITLE
Enable the simd_i16x8_q15mulr_sat_s test on AArch64

### PR DESCRIPTION
**Pull request by akirilov-arm**

### CODE REVIEW

File: cranelift/codegen/src/isa/aarch64/lower_inst.rs [hunk 1]
```diff
@@ -1650,8 +1650,6 @@ pub(crate) fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
             panic!("table_addr should have been removed by legalization!");
         }
 
-        Opcode::ConstAddr => unimplemented!(),
```
Comment (cfallin) [inline cranelift/codegen/src/isa/aarch64/lower_inst.rs:1653]: The consolidation of all of the `unimplemented!()` cases into the wildcard match below, here and in the other `match`es (s390x, x64, interpreter), is definitely valid, but IMHO at least, listing the unimplemented opcodes explicitly serves a useful purpose: it ensures that we don't miss an opcode when we go to eventually implement functionality. Tests also ensure this in a way, but that's more indirect (relies on 100% test coverage); and it is also useful sometimes to actually see what is missing.

Open to other views on this, though; especially interested in what @abrown @jlb6740 @uweigand think, as other backend-focused folks.

**Review by cfallin — COMMENTED** [review]
Implementation LGTM, thanks! Very convenient that aarch64 has an instruction just for this :-)

One thought below on a cleanup you've done -- want to see what others think on this too.

File: cranelift/codegen/src/isa/aarch64/lower_inst.rs (hunk 1 above)
Comment (cfallin) [reply cranelift/codegen/src/isa/aarch64/lower_inst.rs:1653]: (And `ConstAddr` where I placed this line-comment maybe isn't the best example, but e.g. all the SIMD loads were listed out in s390x, whereas an implementer would now likely go through a run-tests, fix-one-panic, retry loop to find all that's needed.)

Comment (uweigand) [reply cranelift/codegen/src/isa/aarch64/lower_inst.rs:1653]: Agreed, I'd also prefer unimplemented opcodes to be listed explicitly, for the very reason you give.

Comment (akirilov-arm) [reply cranelift/codegen/src/isa/aarch64/lower_inst.rs:1653]: @cfallin I based this approach on [the x64 backend](https://github.com/bytecodealliance/wasmtime/blob/main/cranelift/codegen/src/isa/x64/lower.rs#L6002), but I don't mind reverting to the previous one.

Comment (akirilov-arm) [reply cranelift/codegen/src/isa/aarch64/lower_inst.rs:1653]: Also, with Rust there is a quick hack to check what is missing - just comment out the wildcard case and try to rebuild :-).

Comment (cfallin) [reply cranelift/codegen/src/isa/aarch64/lower_inst.rs:1653]: Ah, yes, we are not consistent with the above in `x64`; thanks for pointing that out! I just created #3037 (and this was a good exercise as I learned about a few opcodes we hadn't filled in yet).

Comment (akirilov-arm) [reply cranelift/codegen/src/isa/aarch64/lower_inst.rs:1653]: Speaking of consistency - notice that both in the AArch64 and the s390x backends some matches use `panic!()` for unimplemented operations, while others - `unimplemented!()`. While it doesn't change program behaviour much, IMHO all should use the latter, so I can clean that up instead - open to opinions (from @uweigand as well).

Comment (cfallin) [reply cranelift/codegen/src/isa/aarch64/lower_inst.rs:1653]: Ah, yes, that would be a welcome cleanup as well; thanks!

Comment (uweigand) [reply cranelift/codegen/src/isa/aarch64/lower_inst.rs:1653]: Agreed!

**Comment by akirilov-arm** [comment]
I would like to mention one point as a follow-up to [the discussion](https://github.com/bytecodealliance/wasmtime/pull/2982#discussion_r651773002) in PR #2982 - the new IR operation I am introducing is technically expressible as a combination of existing ones. [This comment](https://github.com/WebAssembly/simd/pull/365#issuecomment-700964442) from the WebAssembly SIMD operation proposal gives an idea how, but note that pattern-matching would be further complicated by the fact that addition is commutative. Overall, I think that in this case the balance is in favour of introducing a new IR operation.

**Comment by cfallin** [comment]
I agree, the in-terms-of-exsting-ops form is 8 Wasm ops, so this is reasonable to create a new opcode for, I think.

**Review by cfallin — APPROVED** [review]
LGTM, thanks!

**Pull request merged**